File: data/transcoder_evaluation_gfg/c/CHECK_LARGE_NUMBER_DIVISIBLE_4_NOT.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

bool f_gold ( char str [] ) {
  int n = strlen(str);
  if ( n == 0 ) return false;
  if ( n == 1 ) return ( ( str [ 0 ] - '0' ) % 4 == 0 );
  int last = str [ n - 1 ] - '0';
  int second_last = str [ n - 2 ] - '0';
  return ( ( second_last * 10 + last ) % 4 == 0 );
}


bool f_filled ( char str [] ) {}

int main(void) {
    int n_success = 0;
    char param0[][100] = {"PjAFZXQgN","12325195609714","00101111101","xOtbXZiA","980","000000100","zFacc W","8","110011","afiutekeSfYrX"};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i]) == f_gold(param0[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}

File: data/transcoder_evaluation_gfg/go/CHECK_LARGE_NUMBER_DIVISIBLE_4_NOT.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(str []byte) bool {
	var n int = int(len(str))
	if n == 0 {
		return false
	}
	if n == 1 {
		return (str[0]-byte('0'))%4 == 0
	}
	var last int = int(str[n-1] - byte('0'))
	var second_last int = int(str[n-2] - byte('0'))
	return (second_last*10+last)%4 == 0
}
//TOFILL
func main() {
	var (
		n_success int           = 0
		param0    [10][]byte = [10][]byte{func() []byte {
			var t []byte
			copy(t[:], ([]byte)("PjAFZXQgN"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("12325195609714"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("00101111101"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("xOtbXZiA"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("980"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("000000100"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("zFacc W"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("8"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("110011"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("afiutekeSfYrX"))
			return t
		}()}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i][:]) == f_gold(param0[i][:]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}
